functions/uploads.py

```diff
@@ -129,25 +146,32 @@ def do_copy(urls):
         # NOTE: this needs to happen after TrialMetadata.patch_assays
         # in order to avoid violating a foreign-key constraint on the trial_id
         # in the event that this is the first upload for a trial.
-        for artifact_metadata, additional_metadata in downloadable_files:
-            print(f"Saving metadata to downloadable_files table: {artifact_metadata}")
-            with saved_failure_status(job, session):
-                DownloadableFiles.create_from_metadata(
-                    trial_id,
-                    job.upload_type,
-                    artifact_metadata,
-                    additional_metadata=additional_metadata,
-                    session=session,
-                    commit=False,
-                )
+        def create_downloadable_file(args):
+            artifact_metadata, additional_metadata = args
+            logger.debug(
+                f"Saving metadata to downloadable_files table: {artifact_metadata}"
+            )
+            DownloadableFiles.create_from_metadata(
+                trial_id,
+                job.upload_type,
+                artifact_metadata,
+                additional_metadata=additional_metadata,
+                session=session,
+                commit=False,
+            )
+
+        with ThreadPoolExecutor(THREADPOOL_THREADS) as executor, saved_failure_status(
+            job, session
+        ):
+            executor.map(create_downloadable_file, downloadable_files)
 
         # Additionally, make the metadata xlsx a downloadable file
+        _, xlsx_blob = _get_bucket_and_blob(GOOGLE_DATA_BUCKET, job.gcs_xlsx_uri)
+        full_uri = f"gs://{GOOGLE_DATA_BUCKET}/{xlsx_blob.name}"
+        data_format = "Assay Metadata"
+        facet_group = f"{job.upload_type}|{data_format}"
+        logger.info(f"Saving {full_uri} as a downloadable_file.")
         with saved_failure_status(job, session):
-            _, xlsx_blob = _get_bucket_and_blob(GOOGLE_DATA_BUCKET, job.gcs_xlsx_uri)
```

Comment:
not sure why you took this RPC out of `saved_failure_status`

Reply:
That was unintentional - good catch.